device/globals/vm_level.gd

```diff
@@ -29,7 +32,11 @@ func _walk(params, block):
 ### commands
 
 func set_global(params):
-	vm.set_global(params[0], params[1])
+	var result = _diff_pattern.search(params[1])
```

Comment:
Is this maybe a bit generic with `_diff_pattern` and `result`?

```
var calc_operation = _calc_pattern.search(params[1])
if calc_operation:
    vm.set_global_calculate(params[0], calc_operation.get_string(1), calc_operation.get_string(2))
```

or somesuch variable names would be more obvious.

Reply:
Yes, you're right. I will change the names to your suggestions.